pkg/controller/clusterclaim/clusterclaim_controller.go

```diff
@@ -200,8 +200,14 @@ func (r *ReconcileClusterClaim) Reconcile(request reconcile.Request) (result rec
 
 	logger = logger.WithField("cluster", clusterName)
 
+	poolLifetime, err := r.clusterPoolLifetimeForClaim(claim, logger)
+	if err != nil {
+		logger.Log(controllerutils.LogLevel(err), "error getting ClusterPool")
+		return reconcile.Result{}, err
+	}
+
 	// Delete ClusterClaim after its lifetime elapses
-	if lifetime := claim.Spec.Lifetime; lifetime != nil {
+	if lifetime := getClaimLifetime(poolLifetime, claim.Spec.Lifetime); lifetime != nil {
```

Comment:
Am I correct that a user can view their ClusterClaim, they are able to set a lifetime on it, see the lifetime that they set, but that may not be the actual lifetime in use at all? If so I fear someone is going to get a nasty surprise from this at an inopportune time. 

I think this plays into the above. I think ClusterPool.Spec.ClaimLifetime should be a default, but the Claim can always override. A claim created without a lifetime, when the pool has one, should be updated to include that value as part of reconciliation so the user can always see the actual value in play by looking at the claim.

WDYT?

Reply:
> Am I correct that a user can view their ClusterClaim, they are able to set a lifetime on it, see the lifetime that they set, but that may not be the actual lifetime in use at all? If so I fear someone is going to get a nasty surprise from this at an inopportune time.

i think we can add a status field to provide the accurate lifetime of the claim which will incorporate both the sources.